src/main/java/core/basesyntax/service/implementations/FruitShopServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.service.implementations;
+
+import core.basesyntax.exceptions.FruitsQuantityException;
+import core.basesyntax.exceptions.TransactionException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.strategy.DataHandlerStrategy;
+import java.util.List;
+
+public class FruitShopServiceImpl implements FruitShopService {
+    private final DataHandlerStrategy dataHandlerStrategy;
+
+    public FruitShopServiceImpl(DataHandlerStrategy dataHandlerStrategy) {
+        this.dataHandlerStrategy = dataHandlerStrategy;
+    }
+
+    // add all information to Storage depending on transactions
```

Comment:
don't add simple comments to your code. If you want to describe sth - use Javadoc